### ISSUE TITLE
Is there a limitation to class methods with default parameters?

### ISSUE THREAD

**Issue by International** [waterlink/mocks.cr#25]
Hi, I'm trying to mock the `Process.run` class method, but with little succes, I've tried the following:

```
require "./spec_helper"

Mocks.create_mock Process do
  mock self.run(command)
  # mock self.run(command, args, env, clear_env, shell, input, output, error, chdir)
end

describe "it should work" do
  it "should mock" do
    allow(Process).to receive(self.run("potato")).and_return(Process::Status.new(0))

    puts "r1: #{Process.run("potato").inspect}"
  end

end
```

this outputs `r1: #<Process::Status:0x1090e2e60 @exit_status=32512>` , so the mock does not get called. I've also tried to have the mock defined with all parameters, such as:
```
Mocks.create_mock Process do
  # mock self.run(command)
  mock self.run(command, args, env, clear_env, shell, input, output, error, chdir)
end
```
but I still get `exit_status=32512`. Is there something I'm doing wrong?

**Comment by waterlink** [comment]
Yeah, it never worked properly with default parameters. And it would require some terrible hacks to make it work with them.

There is a best practice in mocking:

> Do not mock what you do not own.

I would recommend reading this article: https://8thlight.com/blog/eric-smith/2011/10/27/thats-not-yours.html

So you could escape the problem entirely if you had your own object that would wrap a call to `Process.run`. BTW, it could have a very expressive API that matches your business domain.
